js/client-mainmenu.js

```diff
@@ -114,7 +125,68 @@
 				);
 			}
 		},
+		createLoginFormObserver:function (self) {
+			const config = {attributes: true, childList: true, subtree: true};
+			const observer = new MutationObserver(function () {
+				if (window.app.user.get("named") && window.Storage.prefs("page") == "challenge") {
+					window.Storage.prefs("page", "testclient");
```

Comment:
?

Reply:
the code check when the user log in from the link and then activate the user login event

Reply:
I'm asking what's with the `Storage.prefs('page')` call. It doesn't appear to be needed? If it is - why is it hardcoded?

Reply:
there was an issue that trigger challenge after user log in from normal game link and not from the challenge link
so i needed to check if the user logged in after inserting challenge link and not normal link

Reply:
Could you not just check the URL for that information? The `challenge-[]-[]` string should still be there and would be useful for that.